go/pkg/digest/digest.go

```diff
@@ -147,16 +147,8 @@ func NewFromFile(path string) (Digest, error) {
 // NewFromReader computes a file digest from a reader.
 // It returns an error if there was a problem reading the file.
 func NewFromReader(r io.Reader) (Digest, error) {
-	// 32KiB is the buffer size used by io.Copy.
-	return NewFromReaderWithBuffer(r, make([]byte, 32*1024))
-}
-
-// NewFromReaderWithBuffer computes a file digest from a reader, using the
-// given buffer.
-// It returns an error if there was a problem reading the file.
-func NewFromReaderWithBuffer(r io.Reader, buf []byte) (Digest, error) {
 	h := HashFn.New()
-	size, err := io.CopyBuffer(h, r, buf)
+	size, err := io.Copy(h, r)
```

Comment:
Why did you rever these?

HashFn doesn't implement `ReadFrom` [1], and your implemented fileSource doesn't implement `WriteTo`.

1: https://golang.org/pkg/crypto/#Hash

Reply:
Updated this code to use a global pool of buffers. Now packages other than `cas` also benefit from this.

Reply:
Sgtm